### ISSUE TITLE
Terminal env settings not reflected in terminals created using VSCode API

### ISSUE THREAD

**Issue by karrtikr** [microsoft/vscode#192943]
Does this issue occur when all extensions are disabled?: Yes/No

Version: 1.83.0-insider
Commit: 03671d332bf6a34782d5adc64086bc8e083a1590
Date: 2023-09-12T15:02:37.804Z
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36

Steps to Reproduce:

1. Configure fish profile to also set an environment variable:
```json
        "fish": {
            "path": "fish",
            "env": {
                "fish_history": "history"
            }
        },
```
2. Select this as default terminal profile
3. Create a terminal and print variables, notice `fish_history` is present:
![image](https://github.com/microsoft/vscode/assets/13199757/237011c4-aefd-48c2-9383-85d35a337722)

4. Create a terminal using `vscode.window.createTerminal()` API, notice many variables including this are not present:
![image](https://github.com/microsoft/vscode/assets/13199757/c9e1656d-1c90-4ca7-b72d-a152dba4b197)

For Python extension, this leads to "Python" terminals being different from the selected terminal: https://github.com/microsoft/vscode-python/issues/21960. Created on behalf of @kyochikuto.

**Comment by Tyriar** [comment]
I think this is by design because you're launching a shell, not a profile. Unless you don't specify a `shellPath` in which case I haven't really thought about what's meant to happen.

**Comment by kyochikuto** [comment]
I believe we're always launching a `shell`, but with `terminal.integrated.profiles.linux` we just customize as to how we're doing it. Right now I can manually hit `New Terminal` and then do `python script.py` to have the profile settings set during execution but it's somehow cumbersome.

And if it is by design, I'm curious as to why and what are the benefits (since it seems counter-intuitive to me IMHO)?

**Comment by kyochikuto** [comment]
@karrtikr What is your take? From what I understand of @Tyriar comments, Python-ext is more appropriate to implement this, correct?

**Comment by karrtikr** [comment]
> I think this is by design because you're launching a shell, not a profile

I would expect the behavior of `vscode.window.createTerminal()` to be same as when clicking:

![image](https://github.com/microsoft/vscode/assets/13199757/7fb03f7f-0c11-4577-86f7-7012213c2f29)
when opening a new terminal. I think it was this way before the concept of terminal profiles got introduced.

@Tyriar How do we launch such a terminal using vscode API?

**Comment by Tyriar** [comment]
@karrtikr do you explicitly set the shellPath currently?

**Comment by karrtikr** [comment]
Nope, just the name.

**Comment by Tyriar** [comment]
> I think it was this way before the concept of terminal profiles got introduced.

This is right, profiles launching from the extension host was just never considered. 

Thinking about it now the way I would expect it to work is for all props of the default profile to be picked up if `shellPath` and `shellArgs` are undefined. I'm a little concerned if we just turn on linking profiles details as it might break some extensions, but I guess that's the right behavior.

@karrtikr do you have bandwidth to look into implementing this? It's probably just a matter of calling into `ITerminalProfileResolverService` when extension terminals get initialized.

**Comment by kyochikuto** [comment]
I actually need this to prevent VS Code from cluttering my shell history, otherwise my history is filled up with entries such as `source /some/path/.venv/bin/activate.fish` and `/some/path/.venv/bin/python /some/path/src/myscript.py` or subsequent commands I tried for that specific project which are irrelevant as my shell history in my general day to day usage! 

Setting `terminal.integrated.shellIntegration.history` to `0` did not have any effect, at least on my `fish` shell, not sure if it's a bug or I'm missing something! Nor did having `fish_history=` in a `.env` file and opting into the newly introduced `"python.experiments.optInto": [
        "pythonTerminalEnvVarActivation"
    ]`.

**Comment by Tyriar** [comment]
@kyochikuto thanks for the info, I created https://github.com/microsoft/vscode-python/issues/21998 to hopefully solve the issue you're having regardless of this ask. You can also check if `TERM_PROGRAM` = `vscode` ad handle it specially vscode's terminal specially in your shell init which may be helpful.

**Comment by karrtikr** [comment]
> Nor did having fish_history= in a .env file and opting into the newly introduced "python.experiments.optInto": [ "pythonTerminalEnvVarActivation" ]

This should be fixed with https://github.com/microsoft/vscode-python/pull/22004, I've opened https://github.com/microsoft/vscode-python/issues/22005 on your behalf to verify this.

**Comment by kyochikuto** [comment]
This is indeed fixed and I can properly set env vars for my fish shell now, thank you for your time and attention!
